paddle/scripts/travis/build_doc.sh

```diff
@@ -1,15 +1,18 @@
 #!/bin/bash
+set -e
+
+# Create the build directory for CMake.
+mkdir -p $TRAVIS_BUILD_DIR/build
+cd $TRAVIS_BUILD_DIR/build
 
-# Add set -e, cd to directory.
-source ./common.sh
 # Compile Documentation only.
-cmake .. -DCMAKE_BUILD_TYPE=Debug -DCMAKE_Fortran_COMPILER=/usr/bin/gfortran-4.8 -DWITH_GPU=OFF -DWITH_DOC=OFF -DWITH_STYLE_CHECK=OFF ${EXTRA_CMAKE_OPTS}
+cmake .. -DCMAKE_BUILD_TYPE=Debug -DWITH_GPU=OFF -DWITH_DOC=OFF -DWITH_STYLE_CHECK=OFF
 mkdir output
 make -j `nproc`
```

Comment:
Paddle is using `Sphinx` to generate documentation. `Sphinx` will load all Paddle's python package, `py_paddle` and `paddle`. Part of `py_paddle` is not pure Python. It is currently using `SWIG` to generate and linking all Paddle libraries.

So, we must create `py_paddle` and `paddle` Python packages first(the first `cmake`), and install them to current Python interpreter. And then let `sphinx` generate all documentations(the second `cmake`).

If we use C-API to expose Paddle library, then `paddle` Python packages could be pure `Python`. And then, we could just use `sphinx` to generate documentation.